src/checkers/inference/solver/MaxSat2TypeSolver.java

```diff
@@ -132,6 +126,6 @@ public InferenceSolution solve() {
         }
 
 
-        return new DefaultInferenceSolution(result);
+        return new DefaultInferenceResult(result, new HashSet<>());
```

Comment:
For legacy solvers, I assumed there is always solution thus explanation is always an empty map.

Reply:
Agree. Already did that.